cstore_fdw.c

```diff
@@ -300,6 +301,9 @@ CStoreProcessUtility(Node * parseTree, const char *queryString,
 		List *allTablesList = truncateStatement->relations;
 		List *cstoreTablesList = FindCStoreTables(allTablesList);
 		List *otherTablesList = list_difference(allTablesList, cstoreTablesList);
+
+		CheckTruncatePermissions(cstoreTablesList);
```

Comment:
> (1) PostgreSQL's ExecuteTruncate() seems to do permission check & truncation while holding an AccessExclusiveLock, so I guess permissions cannot change after permission check and before end of TRUNCATE.

I'm not particularly concerned about that race. But we really *should* take an AccessExclusiveLock during the check (or just acquire one before), because otherwise there's lock-upgrade issues.

>  (2) Not caused by this change, but PostgreSQL grabs a lock on all relations before starting truncating the first table. In cstore_fdw we lock, truncate, release tables one by one.

I don't think we "release" locks (that's what you mean, right) at all?